### PR TITLE
tools/promtail.sh: add DATAROOT option

### PR DESCRIPTION
**Pull request by daixiang0**
Ensure docker data root is correct, if not, promtail can not get any meaningful data.

### CODE REVIEW

**Comment by gouthamve** [comment]
@DanCech

**Comment by DanCech** [comment]
This won't work as-is because the yaml section doesn't support variable replacement (by design), it would need to be replaced via the sed section at the bottom.

It should also be templated in the ksonnet so we can generate the yaml from the ksonnet by specifying data root as `<dataRoot>` in the ksonnet config.

**Comment by daixiang0** [comment]
I think that tools/promatail.sh can make users use promatail easier in k8s cluster env, it is good to support custom data root.

**Comment by daixiang0** [comment]
@DanCech  any input?

**Comment by DanCech** [comment]
I agree that supporting it is a good idea, but it needs to be done by updating the ksonnet to accept it as a parameter so that we can use that to generate the yaml for promtail.sh rather than it being a one-off.

**Comment by daixiang0** [comment]
Let's do it.

**Comment by daixiang0** [comment]
@DanCech hi, please review it.

File: tools/promtail.sh [hunk 1]
```diff
@@ -134,7 +135,7 @@ spec:
           path: /var/log
         name: varlog
       - hostPath:
-          path: /var/lib/docker/containers
+          path: dataroot/containers
```
Comment (DanCech) [inline tools/promtail.sh]:
```suggestion
          path: <dataroot>/containers
```

File: production/ksonnet/README.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ local promtail = import 'promtail/promtail.libsonnet';
 promtail + {
   _config+:: {
     namespace: 'loki',
+    dataroot: '/var/lib/docker',
```
Comment (DanCech) [inline production/ksonnet/README.md]: I think this setting should be moved to inside the `promtail_config` block.

Comment (daixiang0) [reply production/ksonnet/README.md]: sure, all done.

**Comment by DanCech** [comment]
I made a couple of adjustments but haven't had time to test them yet.  Let me know if they don't make sense to you.

**Comment by daixiang0** [comment]
i test it and work.

**Comment by daixiang0** [comment]
Do not merge into master branch?

**Comment by daixiang0** [comment]
@gouthamve hi, i do not find this in master after merge.

**Comment by daixiang0** [comment]
any update?

**Pull request merged**